src/audio/copier/copier.c

```diff
@@ -457,9 +464,17 @@ static int create_dai(struct comp_dev *parent_dev, struct copier_data *cd,
 
 		break;
 	case ipc4_dmic_link_input_class:
-		dai.type = SOF_DAI_INTEL_DMIC;
-		dai.is_config_blob = true;
-		type = ipc4_gtw_dmic;
+		{
+			dai.type = SOF_DAI_INTEL_DMIC;
+			dai.is_config_blob = true;
+			type = ipc4_gtw_dmic;
+
+			int ret = ipc4_find_dma_config(&dai, (uint8_t *)copier->gtw_cfg.config_data,
+							copier->gtw_cfg.config_length);
+			if (ret != 0)
+				return -EINVAL;
+			dai.out_fmt = (void *)&(copier->out_fmt);
+		}
```

Comment:
same here, declare 'ret' at the top level and avoid indentation change.

Reply:
fixed